src/clj/puppetlabs/http/client/async.clj

```diff
@@ -175,12 +176,13 @@
   ([opts :- common/RawUserRequestOptions
     callback :- common/ResponseCallbackFn
     client :- HttpAsyncClient]
-    (request-with-client opts callback client nil nil))
+    (request-with-client opts callback client nil nil true))
   ([opts :- common/RawUserRequestOptions
     callback :- common/ResponseCallbackFn
     client :- HttpAsyncClient
     metric-registry :- (schema/maybe MetricRegistry)
-    metric-namespace :- (schema/maybe schema/Str)]
+    metric-namespace :- (schema/maybe schema/Str)
+    use-url-metrics :- (schema/maybe schema/Bool)]
```

Comment:
I don't have a super-strong opinion about this.  I'd probably go with just keeping the original 2 arities in place for backward-compatibility and adding a third one with an extra non-optional `enable-url-metrics?` as the last parameter.  I think it would be best to put the new parameter at the end to avoid confusion on the parameter order between the last two arities.  If either you and @rlinehan agree on a different approach, happy to go that way instead, though.